test/test_linalg.py

```diff
@@ -410,6 +410,103 @@ def run_test_case(input, p, dim, keepdim):
                 for ord in ord_settings:
                     run_test_case(input, ord, dim, keepdim)
 
+    @skipCPUIfNoLapack
+    @skipCUDAIfNoMagma
+    @dtypes(torch.float32, torch.float64, torch.complex64, torch.complex128)
+    @dtypesIfCUDA(torch.float32, torch.float64)
+    @precisionOverride({torch.float32: 1e-3})
+    def test_cond(self, device, dtype):
+        def run_test_case(input, ord):
+            result = torch.linalg.cond(input, ord)
+            result_numpy = np.linalg.cond(input.cpu().numpy(), ord)
+            self.assertEqual(result, result_numpy, rtol=1e-2, atol=self.precision)
+
+            # test out= variant
+            out = torch.empty_like(result)
+            ans = torch.linalg.cond(input, ord, out=out)
+            self.assertEqual(ans, out)
+            self.assertEqual(ans, result)
+
+        norm_types = [1, -1, 2, -2, inf, -inf, 'fro', 'nuc', None]
+        input_sizes = [(3, 3), (2, 3, 3, 3)]
+        for input_size in input_sizes:
+            input = torch.randn(*input_size, dtype=dtype, device=device)
+            for ord in norm_types:
+                # frobenius norm not supported for complex tensors
+                if dtype.is_complex and ord == 'fro':
+                    continue
+                run_test_case(input, ord)
+
+        # test for singular input
+        a = torch.eye(3, dtype=dtype, device=device)
+        a[-1, -1] = 0  # make 'a' singular
+        for ord in norm_types:
+            # frobenius norm not supported for complex tensors
+            if dtype.is_complex and ord == 'fro':
+                continue
+            run_test_case(a, ord)
+
+    # TODO: once "inverse_cuda" supports complex dtypes, they shall be added to above tests
```

Comment:
https://github.com/pytorch/pytorch/pull/45034 got merged. This needs to be updated.